packages/search-service/src/search-mapping/helpers.ts

```diff
@@ -0,0 +1,49 @@
+const fetch = require('node-fetch');
+global.Headers = fetch.Headers;
+
+class APITemplateHelper {
+    private static SearchGroupHelperInstance: APITemplateHelper;
+    constructor() { }
+    public static templateHelper() {
+        if (!APITemplateHelper.SearchGroupHelperInstance) {
+            APITemplateHelper.SearchGroupHelperInstance = new APITemplateHelper();
+        }
+        return APITemplateHelper.SearchGroupHelperInstance;
+    }
+
+    public objectStringMapper(object: any, param: string) {
+        let indexProp = param.replace(/\[(\w+)\]/g, '.$1');
+        indexProp = indexProp.replace(/^\./, '');
+        let indexList = indexProp.split('.');
+        indexList.forEach((element: any) => {
+            if (element in object) {
+                object = object[element];
+            } else {
+                return;
+            }
+        });
+        return object;
+    }
+
+    public fetchApi(apiUrl: any, query: any, variables: any, mode: any) {
+        let headers = new Headers();
+        let body = JSON.stringify({
+            query,
+            variables
+        });
+        headers.append(`Authorization`, `${process.env.GATEWAY_AUTH_TOKEN}`);
```

Comment:
You cannot use the Gateway Auth token here, the Gateway Auth toke only works for One Platform API Gateway. the user can have their APIs hosted outside the API Gateway, so we need to get the Authorization Header value from the user and store it in the config.